staging/src/k8s.io/apiserver/pkg/endpoints/filters/audit.go

```diff
@@ -56,7 +56,11 @@ func WithAudit(handler http.Handler, sink audit.Sink, policy policy.Checker, lon
 		}
 
 		ev.Stage = auditinternal.StageRequestReceived
-		processAuditEvent(sink, ev, omitStages)
+		if !processAuditEvent(sink, ev, omitStages) {
```

Comment:
Only reject request on stage `StageRequestReceived`, It is not consistent ignoring other stags audit failure.

Reply:
I doesn't make sense to reject request at later stages (at least for mutating requests), as the request will already have been processed at this point.

Reply:
Yes，it has been processed. But also without audit logs of these stages. So what i want to say is this pr does not handle `lost audit events` completely.

Reply:
With my change, an auditor can get a guarantee that if there was no event with RequestReceived stage, there was no change to apiserver state and the caller will have a guarantee that if the request failed, the state wasn't changed.

With what you are suggesting, auditor will still get the guarantee that no event implies no change, but you will take away that guarantee from the caller. When request fails, you don't really know what happened.

One change I think might make sense here would be to reject non-mutating requests upon failures on other stages. WDYT?

Reply:
After some thought I don't like the idea of rejecting requests inconsistently - with audit policy configured to omit RequestReceived stage this would mean that audit backend issue will block all non-mutating requests, but allow mutating ones.

Reply:
Apologies for being unclear. Let me try to clarify what I meant above.

I think this check should be applied only at `RequestReceived` stage. Later on, a failure to audit log anything shouldn't affect apiserver response, because the request was already processed, possibly modifying etcd state. Now, one could argue that there is no harm in rejecting the request if it was non-mutating, but failed to be audit logged e.g. at `ResponseCompleted` stage. However, this would lead to some weirdness, when `RequestReceived` stage is omitted and audit backend has problems: mutating requests would ignore audit logging failures and simply work, while read only requests would be rejected because of audit backend problems.